### DISCUSSION TITLE
Dashboard development, Git and Pull requests

### DISCUSSION THREAD

**Discussion by DannyArr** [grafana/grafana#57550] [General]
**The problem**: we want to allow all developers the ability to contribute to our dashboards but we've found that this leads to an unstructured mess of team specific dashboards that don't look at the bigger picture or tie in with other upstream or downstream teams very well at all.

So we are currently redesigning our layout to have a hierarchy of linked dashboards that make sense and allow a user to get a good idea of the entire estate quickly and easily. But then we have to lock down the dashboards and the developers can no longer create or edit them.

What would be great is if developers could go through a Pull Request process to ensure our standards are being adhered to. Have some kind of Approval process. To do this we have to use something like GitHub. We tried looking at Graffonet but a) it doesn't have an Azure Monitor target that we could find and B) we really don't want to be creating and editing dashboards in code when Dashboards are really all about visualisations and getting the devs to learn yet another thing may not go down so well.

So I thought I'd open a discussion around the idea of having dashboards maintained in GitHub. You could then have a drop down in the UI to select your branch. If you're in the main branch you may be unable to edit or create anything according to permissions. But you could select or create a different branch which would then to the dashboards and folders in that branch. You make your changes there, submit a PR from the UI which can be approved and merged into Main by those with the power to do so.

Maybe main could be copied to the local database on every merge as well, to reduce the latency of having to retrieve them from GiHub every time. Maybe some caching.

Maybe the problem has already been solved via a different method. But having this kind of Integration with Git seems like it could vastly improve the development process and quality. Thoughts?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
